File: [0703][Easy][kth-largest-element-in-a-stream]/kth_largest_element_in_a_stream.java
import java.util.PriorityQueue;
class KthLargest {
    final PriorityQueue<Integer> q ;
    final int k;
    public KthLargest(int k, int[] nums) {
        this.k = k;
        q = new PriorityQueue<Integer>(k);
        for(int i: nums) {
            add(i);
        }
    }
    
    public int add(int val) {
        if(q.size() < k) {
            q.offer(val);
            
        }
        else if(q.peek() < val) {
            q.poll();
            q.offer(val);
        }
        return q.peek();
    }
}
public class kth_largest_element_in_a_stream{
    public static void main(String args[]){
        int k = 3;
        int nums[] = {4,5,8,2};
        KthLargest obj = new KthLargest(k, nums);
        int param_1 = obj.add(3);
        param_1 = obj.add(5);
        param_1 = obj.add(10);
        param_1 = obj.add(9);
        System.out.println("In java code,last return value is :" + param_1);
    }
}
/**
 * Your KthLargest object will be instantiated and called as such:
 * KthLargest obj = new KthLargest(k, nums);
 * int param_1 = obj.add(val);
 */

File: [0703][Easy][kth-largest-element-in-a-stream]/kth-largest-element-in-a-stream.py
class KthLargest:
    def __init__(self, k, nums):
        self.nums = nums
        self.k = k
        self.nums.sort(reverse = True)
        while len(self.nums) > k:
            self.nums.pop()

    def add(self, val):
        self.nums.append(val)
        self.nums.sort(reverse = True)
        if len(self.nums) > self.k:
            self.nums.pop()
        return self.nums[-1]
    
def main():
    k = 3
    nums = [4,5,8,2]
    obj = KthLargest(k, nums)
    param_1 = obj.add(3)
    param_1 = obj.add(5)
    param_1 = obj.add(10)
    param_1 = obj.add(9)
    print("last return value is "+str(param_1));
    
if __name__ =='__main__':
    main()      
# Your KthLargest object will be instantiated and called as such:
# obj = KthLargest(k, nums)
# param_1 = obj.add(val)